enterprise/server/cmd/ci_runner/main.go

```diff
@@ -141,13 +143,26 @@ func RunAllActions(ctx context.Context, cfg *config.BuildBuddyConfig, im *initMe
 	for _, action := range cfg.Actions {
 		startTime := time.Now()
 
-		if !matchesAnyTrigger(action, *triggerEvent, *triggerBranch) {
-			log.Debugf("No triggers matched for %q event with target branch %q. Action config:\n===\n%s===", *triggerEvent, *triggerBranch, actionDebugString(action))
+		if *actionName != "" {
+			if action.Name != *actionName {
+				continue
+			}
+		} else if !matchesAnyTrigger(action, *triggerEvent, *triggerBranch) {
+			log.Printf("No triggers matched for %q event with target branch %q. Action config:\n===\n%s===", *triggerEvent, *triggerBranch, actionDebugString(action))
 			continue
 		}
 
+		iid := ""
+		// Respect the invocation ID flag only when running a single action
+		// (via ExecuteWorkflow).
+		if *actionName != "" {
+			iid = *invocationID
+		}
+		if iid == "" {
+			iid = newUUID()
+		}
```

Comment:
Why is this necessary?

EDIT: OK, I guess it's because you want to ensure 1-1 mapping between actions and invocation IDs. Then why not return an error or log.Fatalf in this case since it indicates bad input?

Reply:
Done (checked the flags at the start of `main`)